src/app/api/utils/auth.tsx

```diff
@@ -81,6 +81,31 @@ export const authOptions: AuthOptions = {
   },
   providers: [fxaProviderConfig],
   callbacks: {
+    redirect({ url, baseUrl }) {
+      // Do not pass un-parseable URLs to next-auth, @see MNTOR-3029
+      // Preserve default behavior otherwise. @see https://next-auth.js.org/configuration/callbacks#redirect-callback
+
+      // Allows relative callback URLs
+      if (url.startsWith("/")) {
+        if (!URL.canParse(`${baseUrl}${url}`)) {
+          logger.warn("Cannot parse redirect url:", {
+            url: `${baseUrl}${url}`,
+          });
+          return baseUrl;
+        }
+        return `${baseUrl}${url}`;
+      }
+      // Allows callback URLs on the same origin
+      else if (new URL(url).origin === baseUrl) {
+        if (!URL.canParse(url)) {
+          logger.warn("Cannot parse redirect url:", { url });
+          return baseUrl;
+        }
```

Comment:
I think `new URL` throws when `url` cannot be parsed, so I think this clause is never hit? (And thus we'd still see the parsing errors in the logs?)

Reply:
Hm OK well it makes sense why it was throwing before... I'm a little worried about changing the behavior here, I guess next-auth throws an error response here now.

I'm going to try making it explicitly return something like:
```
NextResponse.json({ success: false }, { status: 403 })
```
If the URL passed looks absolute and it can't be parsed.

Reply:
Hm OK returning `NextResponse.json` won't work here, I just walked through this with a debugger and see what the problem is. I left a minimal `curl` command in the Jira ticket too.

Reply:
I re-wrote this so `URL.canParse` is called before `new URL(url).origin`.

Reply:
Did you maybe forget to push it? I think I still see it after `new URL`.

Also, I just learned of the static [`URL.parse`](https://developer.mozilla.org/en-US/docs/Web/API/URL/parse_static) method, which doesn't throw but returns `null` for invalid URLs - that might be relevant to your interests.